x-pack/heartbeat/scenarios/basics_test.go

```diff
@@ -34,20 +42,45 @@ func TestSimpleScenariosBasicFields(t *testing.T) {
 				},
 			}), e.Fields)
 
-			// Ensure that all check groups are equal and don't change
-			cg, err := e.GetValue("monitor.check_group")
+			// Ensure that all check groups are equal and don't except across retries
+			cgIface, err := e.GetValue("monitor.check_group")
 			require.NoError(t, err)
-			cgStr := cg.(string)
-			if i == 0 {
-				lastCg = cgStr
-			} else {
-				require.Equal(t, lastCg, cgStr)
+			cg := cgIface.(string)
```

Comment:
I tend to skip that in tests because the verbosity is unnecessary, the worst case is it panics and we get a test failure. It's a little uglier, but I think it makes reading the test significantly easier by not cluttering it up with assertions that aren't getting at the thing we're actually testing.